core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +265,93 @@ public void setConf(Configuration conf) {
     this.conf = conf;
   }
 
+  private boolean insertProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int insertedRecords = connections.run(conn -> {
+        String sqlStatement = JdbcUtil.insertPropertiesStatement(properties);
+
+        try (PreparedStatement sql = conn.prepareStatement(sqlStatement)) {
+          int rowIndex = 0;
+          for (Map.Entry<String, String> keyValue : properties.entrySet()) {
+            sql.setString(++rowIndex, catalogName);
+            sql.setString(++rowIndex, namespaceName);
+            sql.setString(++rowIndex, keyValue.getKey());
+            sql.setString(++rowIndex, keyValue.getValue());
```

Comment:
In Iceberg, we don't use the return values of the `++` operator. It's hard for people to reason about. I think it would be cleaner to use `rowIndex += 4` at the end of the loop and set each argument with `rowIndex + 1`, `rowIndex + 2`, etc.

Also, it may be even easier if you use `Streams.mapWithIndex`:

```java
    Streams.mapWithIndex(
        properties.entrySet().stream(),
        (entry, index) -> {
          sql.setString((int) index + 1, entry.getKey());
          ...
        });
```

Reply:
I had that earlier but @jackye1995  suggested to add ++ instead.